lapack/testlapack/dlangb_bench.go

```diff
@@ -0,0 +1,51 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"testing"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/lapack"
+)
+
+func DlangbBenchmark(b *testing.B, impl Dlangber) {
+	rnd := rand.New(rand.NewSource(1))
+	for _, bm := range []struct {
+		n, k int
+	}{
+		{n: 1000, k: 0},
+		{n: 1000, k: 1},
+		{n: 1000, k: 2},
+		{n: 1000, k: 10},
+		{n: 1000, k: 20},
+		{n: 1000, k: 30},
+		{n: 10000, k: 0},
+		{n: 10000, k: 1},
+		{n: 10000, k: 2},
+		{n: 10000, k: 10},
+		{n: 10000, k: 30},
+		{n: 10000, k: 60},
+		{n: 10000, k: 100},
+	} {
+		n := bm.n
+		k := bm.k
+		lda := 2*k + 1
+		a := make([]float64, n*lda)
+		for i := range a {
+			a[i] = rnd.NormFloat64()
```

Comment:
Is there any merit in looking at small and large values since they impact on `Dlassq`?

Reply:
Good idea. I'll check.

Reply:
Done. Matrices with medium-sized elements are (as expected) the fastest. Excerpt from the benchmark output:
```
BenchmarkDlangb/Frobenius_N=10000_K=10_Small
BenchmarkDlangb/Frobenius_N=10000_K=10_Small-4       	    1778	    776320 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=10_Medium
BenchmarkDlangb/Frobenius_N=10000_K=10_Medium-4      	    2898	    421678 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=10_Big
BenchmarkDlangb/Frobenius_N=10000_K=10_Big-4         	    1339	    888068 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=30_Small
BenchmarkDlangb/Frobenius_N=10000_K=30_Small-4       	     865	   1379806 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=30_Medium
BenchmarkDlangb/Frobenius_N=10000_K=30_Medium-4      	    1004	   1188452 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=30_Big
BenchmarkDlangb/Frobenius_N=10000_K=30_Big-4         	     794	   1512464 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=60_Small
BenchmarkDlangb/Frobenius_N=10000_K=60_Small-4       	     459	   2617403 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=60_Medium
BenchmarkDlangb/Frobenius_N=10000_K=60_Medium-4      	     552	   2171272 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=60_Big
BenchmarkDlangb/Frobenius_N=10000_K=60_Big-4         	     495	   2414113 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=100_Small
BenchmarkDlangb/Frobenius_N=10000_K=100_Small-4      	     286	   4195023 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=100_Medium
BenchmarkDlangb/Frobenius_N=10000_K=100_Medium-4     	     345	   3477300 ns/op
BenchmarkDlangb/Frobenius_N=10000_K=100_Big
BenchmarkDlangb/Frobenius_N=10000_K=100_Big-4        	     328	   3607253 ns/op
```